fn_spamhaus_query/fn_spamhaus_query/components/fn_spamhaus_query_submit_artifact.py

```diff
@@ -0,0 +1,103 @@
+# (c) Copyright IBM Corp. 2010, 2019. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+"""Function implementation"""
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import fn_spamhaus_query.util.selftest as selftest
+from resilient_lib import ResultPayload, RequestsCommon
+from fn_spamhaus_query.util.info_response import STATIC_INFO_RESPONSE
+from fn_spamhaus_query.util.spamhause_helper import *
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'fn_spamhaus_query_submit_artifact"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_spamhaus_query", {})
+        selftest.selftest_function(opts)
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_spamhaus_query", {})
+
+    @function("fn_spamhaus_query_submit_artifact")
+    def _fn_spamhaus_query_submit_artifact_function(self, event, *args, **kwargs):
+        """Function: Function to check IP Address & Domain Names against Spamhaus Database to see whether IP Address or Domain Names appears in Spamhaus block list records or not."""
+        try:
+            # Get the function parameters:
+            spamhaus_query_string = kwargs.get("spamhaus_query_string")  # text
+            spamhause_search_resource = kwargs.get("spamhause_search_resource")  # text
+
+            # Get the app.config parameters:
+            spamhaus_wqs_url = self.options.get("spamhaus_wqs_url")
+            spamhaus_dqs_key = self.options.get("spamhaus_dqs_key")
+
+            log = logging.getLogger(__name__)
+            log.info("spamhaus_query_string: %s", spamhaus_query_string)
+            log.info("spamhause_search_resource: %s", spamhause_search_resource)
+            log.info("spamhaus_wqs_url: %s", spamhaus_wqs_url)
+            log.info("spamhaus_dqs_key: %s", spamhaus_dqs_key)
+
+            yield StatusMessage(
+                "Checking Artifact: {} against Spamhaus block list resource {} database.".format(spamhaus_query_string,
```

Comment:
```suggestion
                u"Checking Artifact: {} against Spamhaus block list resource {} database.".format(spamhaus_query_string,
```

Might want to add the `u` so that the StatusMessage handles unicode

Reply:
ok. done